interface/procedure_tools/labcorp/ereq_form.php

```diff
@@ -745,28 +755,13 @@ function ereqForm($pid, $encounter, $form_id, $reqStr = null, $doDoc = true)
         $DOCUMENT_CATEGORY = $category['id'];
 
         $d = new Document();
-        $d->name = $filename;
-        $d->storagemethod = 0;
-        $d->url = "file://" . $directory . $filename;
-        $d->mimetype = "application/pdf";
-        $d->size = filesize($directory . $filename);
-        $d->owner = $_SESSION['authUserID'];
-        $d->hash = sha1_file($directory . $filename);
-        $d->type = $d->type_array['file_url'];
-        $d->list_id = (int)$form_id;
-        $d->set_foreign_id($pid);
-        $d->persist();
-        $d->populate();
-        $docid = $d->get_id();
+        $docid = $d->createDocument($pid, $DOCUMENT_CATEGORY, $filename, "application/pdf", $mpdfData);
```

Comment:
@sjpadgett ,
Addressed all the issues in this commit. The changes at this spot should work (changed to using the standard createDocument mechanism), although i don't know how to test it.

Reply:
note i just modified this mechanism even further and removed the need for creating the file (sent the mpdf output to a variable).